packages/flutter/lib/src/cupertino/desktop_text_selection.dart

```diff
@@ -0,0 +1,210 @@
+      handleCut: canCut(delegate) ? () => handleCut(delegate) : null,
+      handleCopy: canCopy(delegate) ? () => handleCopy(delegate, clipboardStatus) : null,
+      handlePaste: canPaste(delegate) ? () => handlePaste(delegate) : null,
+      handleSelectAll: canSelectAll(delegate) ? () => handleSelectAll(delegate) : null,
+      selectionMidpoint: selectionMidpoint,
+      lastTapDownPosition: lastTapDownPosition,
+      textLineHeight: textLineHeight,
+    );
+  }
+
+  /// Builds the text selection handles, but desktop has none.
+  @override
+  Widget buildHandle(BuildContext context, TextSelectionHandleType type, double textLineHeight) {
+    return const SizedBox.shrink();
+  }
+
+  /// Gets the position for the text selection handles, but desktop has none.
+  @override
+  Offset getHandleAnchor(TextSelectionHandleType type, double textLineHeight) {
+    return Offset.zero;
+  }
+}
+
+/// Text selection controls that follows Mac design conventions.
+final TextSelectionControls cupertinoDesktopTextSelectionControls =
+    _CupertinoDesktopTextSelectionControls();
+
+// Generates the child that's passed into CupertinoDesktopTextSelectionToolbar.
+class _CupertinoDesktopTextSelectionControlsToolbar extends StatefulWidget {
+  const _CupertinoDesktopTextSelectionControlsToolbar({
+    Key? key,
+    required this.clipboardStatus,
+    required this.endpoints,
+    required this.globalEditableRegion,
+    required this.handleCopy,
+    required this.handleCut,
+    required this.handlePaste,
+    required this.handleSelectAll,
+    required this.selectionMidpoint,
+    required this.textLineHeight,
+    required this.lastTapDownPosition,
+  }) : super(key: key);
+
+  final ClipboardStatusNotifier? clipboardStatus;
+  final List<TextSelectionPoint> endpoints;
+  final Rect globalEditableRegion;
+  final VoidCallback? handleCopy;
+  final VoidCallback? handleCut;
+  final VoidCallback? handlePaste;
+  final VoidCallback? handleSelectAll;
+  final Offset? lastTapDownPosition;
+  final Offset selectionMidpoint;
+  final double textLineHeight;
+
+  @override
+  _CupertinoDesktopTextSelectionControlsToolbarState createState() => _CupertinoDesktopTextSelectionControlsToolbarState();
+}
+
+class _CupertinoDesktopTextSelectionControlsToolbarState extends State<_CupertinoDesktopTextSelectionControlsToolbar> {
+  late ClipboardStatusNotifier _clipboardStatus;
+
+  void _onChangedClipboardStatus() {
+    setState(() {
+      // Inform the widget that the value of clipboardStatus has changed.
+    });
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _clipboardStatus = widget.clipboardStatus ?? ClipboardStatusNotifier();
+    _clipboardStatus.addListener(_onChangedClipboardStatus);
+    _clipboardStatus.update();
+  }
+
+  @override
+  void didUpdateWidget(_CupertinoDesktopTextSelectionControlsToolbar oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.clipboardStatus == null && widget.clipboardStatus != null) {
```

Comment:
Could short-circuit first:
```dart
if (oldWidget.clipboardStatus == widget.clipboardStatus)
  return;
```

In fact I think this could all be a little simpler:

```dart
  void didUpdateWidget(_CupertinoDesktopTextSelectionControlsToolbar oldWidget) {
    super.didUpdateWidget(oldWidget);
    if (oldWidget.clipboardStatus != widget.clipboardStatus) {
      if (_clipboardStatus != null) {
        _clipboardStatus.removeListener(_onChangedClipboardStatus);
        _clipboardStatus.dispose();
      }
      _clipboardStatus = widget.clipboardStatus ?? ClipboardStatusNotifier();
      _clipboardStatus.addListener(_onChangedClipboardStatus);
      if (widget.handlePaste != null) {
        _clipboardStatus.update();
      }
    }
  }
```

Reply:
Thank you!  Much simpler.